### ISSUE TITLE
Accessibility: Language switcher needs to be completely usable with keyboard

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/ibm-dotcom-library#1720]
The language switcher in the footer needs to be completely usable with the keyboard.
Preferably, it should even be "delightfully usable" with the keyboard. :)

Here's a little scenario to show what a "delightful experience" might be like, if it was problem-free.

Steps (note: do all steps in the footer without touching your mouse or trackpad):
1. Go to: https://ibmdotcom-react.mybluemix.net/iframe.html?id=components-footer--default
2. Start typing the tab key all the way until the language switcher has focus.
3. Type space or enter to open the dialog.
**Problem 1: Where is the focus? There should be a clear focus indicator.**
4. Oops, didn't mean to open that - type Esc to close it...
**Problem 2: Esc key should close dialog no matter which part of dialog has focus**
5. Type tab to move focus to the close (X) button.
6. Type space or enter to activate the close button and close the dialog.
7. Oops, didn't mean to close that - type space or enter to open it again...
**Problem 3: Focus should return to the trigger element when a dialog is closed**
8. Type shift+tab to move focus back to the switcher button, and then space or enter to open again.
9. Type tab to focus the close button, and then type tab again.
**Problem 4: Where are you? What has the focus? There should be a clear focus indicator.**
10. Type Enter to try to figure out where you are. Oh! It switched to the Americas page...
**Problem 5: But there's no focus indicator to show where the focus is.**
11. Type tab and note that focus moves to the Search field. Type aru and note that there's only one item (Aruba English) in the list.
**Problem 6: Wouldn't it be really nice to be able to just type enter here to select Aruba English (instead of tab, tab, enter), since there's only one item in the list?**
12. Oops, decided not to switch to Aruba, so type Esc to clear the aru from the search field...
**Problem 7: The dialog closes. When Esc is typed in a search field, the text should be cleared, and you need to call event.preventDefault() and event.stopPropagation() so that the dialog doesn't close. If the user types Esc in the search field when the search field is empty, _then_ you can close the dialog.**
13. So (remember to use only the keyboard!), open the dialog again, navigate to the Americas, and type aru in the search field.
14. This time, type tab to move focus to the clear (X) button, and type space or enter to clear the search field.
15. Type ar. There are 5 items in the list. Type down arrow to select an item in the list...
**Problem 8: Down arrow should navigate to the first item in the list, and when in the list, down and up arrow should be used to navigate through the items, and NOT the tab key, i.e. the list itself is a single tab stop control.**
16. Now imagine that you really wanted to select a language in Europe. You are currently in the Americas. How do you get to Europe using only your keyboard?
**Problem 9: You can't focus the "<- Select region" button using only the keyboard. The only way to go back and select another region with the keyboard is to close the dialog and start again.**

**Comment by carmacleod** [comment]
Also, if you type the tab key past the end of the dialog, the user might want to type shift+tab to go back into the dialog, but it doesn't work. To get out of that situation the user has to type tab again to wrap focus to the top of the dialog.

**Comment by ljcarot** [comment]
Thanks Carolyn!

**Comment by RobertaJHahn** [comment]
Linda creating an issue to get Ram's feedback on the accessibility issues.

**Comment by ljcarot** [comment]
New issue created for Accessibility team to run tests on https://github.com/carbon-design-system/ibm-dotcom-library/issues/1726

**Comment by RobertaJHahn** [comment]
Accessibility testing was finished.  Bugs were opened based on that testing.  The bugs are being prioritized and scheduled in the future sprints.  Closing this issue.
